tests/debugger/test_debugger_pii.py

```diff
@@ -114,49 +112,121 @@
+
+                if check_field_name in fields:
+                    not_found.remove(field_name)
+
+                    if "value" in fields[check_field_name]:
+                        not_redacted.append(field_name)
+
+        error_message = ""
+        if not_redacted:
+            not_redacted.sort()
+            error_message = "Fields not properly redacted: " + "".join([f"{item}, " for item in not_redacted])
+
+        if not_found:
+            not_found.sort()
+            error_message += ". Fields not found: " + "".join([f"{item}, " for item in not_found])
+
+        if error_message != "":
+            raise ValueError(error_message)
 
-        self._validate_pii_keyword_redaction(redacted_keys, line_probe=line_probe)
-        self._validate_pii_type_redaction(redacted_types, line_probe=line_probe)
+    def _validate_pii_type_redaction(self, should_redact_types, line_probe):
+        not_redacted = []
+
+        for probe_id in self.probe_ids:
+            snapshot = self.probe_snapshots[probe_id][0]["debugger"]["snapshot"]
+
+            for type_name in should_redact_types:
+                if line_probe:
+                    type_info = snapshot["captures"]["lines"]["33"]["locals"][type_name]
+                else:
+                    type_info = snapshot["captures"]["return"]["locals"][type_name]
 
-    def setup_pii_redaction_full(self):
-        self._setup("pii")
+                if "fields" in type_info:
+                    not_redacted.append(type_name)
+
+        error_message = ""
+        if not_redacted:
+            not_redacted.sort()
+            error_message = "Types not properly redacted: " + "".join([f"{item}, " for item in not_redacted])
+
+        if error_message != "":
+            raise ValueError(error_message)
+
+    ############ test ############
+    ### method ###
+    def setup_pii_redaction_method_full(self):
+        self._setup()
 
     @missing_feature(context.library < "java@1.34", reason="keywords are not fully redacted")
     @missing_feature(context.library < "dotnet@2.51", reason="keywords are not fully redacted")
     @bug(context.library == "python@2.16.0", reason="DEBUG-3127")
     @bug(context.library == "python@2.16.1", reason="DEBUG-3127")
-    # Ruby requires @irrelevant rather than @missing_feature to skip setup
-    # for this test (which will interfere with the line probe test).
-    @irrelevant(context.library == "ruby", reason="Local variable capture not implemented for method probes")
-    def test_pii_redaction_full(self):
-        self._test(REDACTED_KEYS, REDACTED_TYPES)
+    @missing_feature(context.library == "ruby", reason="Local variable capture not implemented for method probes")
+    def test_pii_redaction_method_full(self):
+        self._assert(REDACTED_KEYS, REDACTED_TYPES)
+
+    ### line ###
+    def setup_pii_redaction_line_full(self):
+        self._setup(line_probe=True)
+
+    @missing_feature(context.library != "ruby", reason="Tests not implemented yet.")
```

Comment:
For Ruby it is not the tests which are not implemented but Ruby DI does not provide the functionality required for the test.

Reply:
I'm not sure I understand, please explain

Reply:
Aaaaah, I think I got it! You mean the description of decorator is not accurate?

Reply:
Fixed here 964550a37b2c4b305fc79ae703482a18fad160bc